### ISSUE TITLE
Capsule does not have 'offset' attribute

### ISSUE THREAD

**Issue by hmeadow** [nalexn/ViewInspector#85]
When attempting to get the offset off of a Capsule shape we receive the error that 'Capsule does not have offset attribute'. Looking at your examples. it appears this should work. Thoughts?

``` swift
Capsule()
 .frame(height: height * heightRatio)
 .offset(x: 0, y: height * -offsetRatio)
```

``` swift
try view.inspect().shape().offset()
```

**Comment by nalexn** [comment]
Hey @hmeadow

SwiftUI offers two `offset` functions: one for an arbitrary `View`, and another one for the `Shape`.

When you apply `offset` to an arbitrary `View`, the result is `some View`.

When you apply `offset` to a `Shape`, the result is still a `Shape` - SwiftUI incapsulates the applied modifier inside the view structure.

ViewInspector also offers two `offset` functions for unwrapping the value from `Shape` and from an arbitrary view.

Here is what happens in your example:

SwiftUI does not have a special `.frame` modifier that would incapsulate in the `Shape`, so the result is an arbitrary view, not a `Shape`. Then you apply `offset` to an arbitrary view.

In your test, you obviously can unwrap the `Shape` with `try view.inspect().shape()`, but subsequent `offset` function is called for the type `Shape`, and your shape indeed does not have incapsulated `offset` - it's plain `Capsule`

Long story short, the fix is to not unwrap the `Shape`: `try view.inspect().offset()`

Alternatively, you can swap `frame` and `offset`. The result will be a modified `Capsule` with a frame, which you can unwrap as a `Shape` and inspect the `offset` for it.

**Comment by hmeadow** [comment]
Hi @nalexn - Thanks for your reply! Unfortunately, your first suggestion did not work. However, we found that having `offset` as the first modifier (before `frame` and `fill`) worked out. Thanks so much!
